pg_ivm.c

```diff
@@ -197,6 +197,10 @@ create_immv(PG_FUNCTION_ARGS)
 
 	parsetree = linitial_node(RawStmt, parsetree_list);
 
+	/* view definition should spcify SELECT query */
+	if (!IsA(parsetree->stmt, SelectStmt))
+		elog(ERROR, "view definition must specify SELECT statement");
```

Comment:
Please use ereport instead of elog for user-facing error message?

Reply:
thank you. I fix it.